components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.common/src/main/java/org/wso2/carbon/apimgt/micro/gateway/common/api/GatewayUtil.java

```diff
@@ -0,0 +1,162 @@
+
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+            String adminName = ServiceReferenceHolder.getInstance().getRealmService().getTenantUserRealm(tenantId)
+                    .getRealmConfiguration().getAdminUserName();
+            Registry configUserRegistry = ServiceReferenceHolder.getInstance().getRegistryService()
+                            .getConfigUserRegistry(adminName, tenantId);
+            String apiLifeCycleConfig = CommonUtil.getLifecycleConfiguration(API_LIFECYCLE, configUserRegistry);
+            Document doc = convertStringToDocument(apiLifeCycleConfig);
+            //Replace default API executor
+            int n = 0;
+            if (doc != null) {
+                n = doc.getElementsByTagName(EXECUTION_TAG).getLength();
+            }
+            for (int i = 0; i < n; i++) {
+                Element element = (Element) doc.getElementsByTagName(EXECUTION_TAG).item(i);
+                if (RE_PUBLISH_ATTRIBUTE.equals(element.getAttribute(FOR_EVENT)) ||
+                        PUBLISH_ATTRIBUTE.equals(element.getAttribute(FOR_EVENT))) {
+                    element.setAttribute(CLASS, CUSTOM_API_EXECUTOR_CLASS);
+                }
+            }
+            //Add new execution for 'Published' state
+            n = 0;
+            if (doc != null) {
+                n = doc.getElementsByTagName(STATE).getLength();
+            }
+            for (int i = 0; i < n; i++) {
+                Element element = (Element) doc.getElementsByTagName(STATE).item(i);
+                Element dataElement;
+                Element newElement;
+                if (PUBLISHED_STATE.equals(element.getAttribute(ID))) {
+                    dataElement = (Element) element.getElementsByTagName(DATA).item(0);
+                    newElement = doc.createElement(EXECUTION_TAG);
+                    newElement.setAttribute(FOR_EVENT, PUBLISH_ATTRIBUTE);
+                    newElement.setAttribute(CLASS, CUSTOM_API_EXECUTOR_CLASS);
+                    dataElement.appendChild(newElement);
+                }
+            }
+            String updatedApiLifeCycleConfig = convertDocumentToString(doc);
+            // ToDo Generalise the fix for automatic xmlns="" namespace (un)declaration
+            updatedApiLifeCycleConfig = updatedApiLifeCycleConfig.replaceAll("xmlns=\"\"", "");
+            CommonUtil.updateLifecycle(API_LIFECYCLE, updatedApiLifeCycleConfig, configUserRegistry,
+                    CommonUtil.getRootSystemRegistry());
+        } catch (XMLStreamException | RegistryException | UserStoreException | TransformerException | SAXException |
+                ParserConfigurationException | IOException e) {
+            throw new OnPremiseGatewayException("An error occurred while overriding default API execution class of " +
```

Comment:
Do we required to log the error? also exception is thrown but stacktrace now throwed upwords (e).

Reply:
It gets logged in the function that calls it. Will add stack-trace to the exception.